### DISCUSSION TITLE
entity not discovered

### DISCUSSION THREAD

**Discussion by RichardWright** [mikro-orm/mikro-orm#3466] [Q&A]
My config code:

`const commonDbOptions = {
        logger: (msg) => logger.info(msg),
        entities: [Recording, RecordingData, RecordingKeyMoment, Replay],
        forceUtcTimezone: true,
        debug: false,
      };

        orm = await MikroORM.init<PostgreSqlDriver>({
          ...commonDbOptions,
          type: 'postgresql',
          clientUrl: process.env?.POSTGRES_CONNECTION_URL,
        });

        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
        const postgresEntityManager = orm.em as PostgresqlEntityManager;
        orm.em = postgresEntityManager;
      `

calling `await orm.discoverEntities();` doesn't help

any ideas?

**Comment by B4nan** [comment]
Provide more details, this is not enough.

>     const postgresEntityManager = orm.em as PostgresqlEntityManager;

There is no such class as `PostgresqlEntityManager`. And you dont need to cast anything here if you pass the generic type arg to the `init` method.

>     orm.em = postgresEntityManager;

This does not make any sense to me - you dont change the `orm.em` type by reassigning it. That is not how TS works.

> calling await orm.discoverEntities(); doesn't help

That happens automatically in the `init` call.
